### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by dirkhh**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

I triaged a few more - some false positives marked as such, and three reasonably obvious fixes.

### Mentions:

@bstoeger - the last one was introduce by you, I think. Can you check my logic just in case I'm missing something subtle here?

### CODE REVIEW

File: core/ostctools.c [hunk 1]
```diff
@@ -97,7 +97,6 @@ void ostctools_import(const char *file, struct dive_table *divetable, struct tri
 	// Read dive's raw data, header + profile
 	if (fseek(archive, 456, 0) == -1) {
 		report_error(failed_to_read_msg, file);
-		free(uc_tmp);
```
Comment (bstoeger) [inline core/ostctools.c:100]: 1) Obviously correct, but that raises the question why `uc_tmp` is allocated on the heap in the first place
2) For better encapsulation, the `free(ostcdive)` calls should probably be `free_dive(ostcdive)`

File: core/parse.c [hunk 2]
```diff
@@ -246,12 +246,13 @@ void dive_end(struct parser_state *state)
 {
 	if (!state->cur_dive)
 		return;
-	if (!is_dive(state))
+	if (!is_dive(state)) {
 		free_dive(state->cur_dive);
-	else
+	} else {
```
Comment (bstoeger) [inline core/parse.c:251]: Yes, that fixes a subtle bug.

File: core/ostctools.c (hunk 1 above)
Comment (dirkhh) [reply core/ostctools.c:100]: I think I implemented both of these suggestions in my force push.
I'd appreciate another look :-)

**Comment by dirkhh** [comment]
now transfer.sh is stalling. I'll merge this and fix the PR artifacts. Grmbl

**Pull request merged**